feat: add restart from YAML option to Flow and Executors (#5488)
jina-ai/jina: docs/fundamentals/flow/create-flow.md

e.set()  # set event and stop (unblock) the Flow
```

(restart-flow)=
## Restart

> While developing your Flow, it can be useful to have the Flow be restarted automatically as you keep applying changes to the Flow YAML.

Reviewer: ```suggestion
While developing your Flow, it can be useful to have it restart automatically as you change the Flow YAML.
```

> For this you can use the `restart` argument for the Flow so that it watches changes in the configuration YAML code and makes sure that the Flow is restarted with the new configuration.

Reviewer: ```suggestion
For this you can use the Flow's `restart` argument to restart the Flow with the updated configuration every time you change the YAML configuration.
```

````{admonition} Caution
:class: caution

> This feature is thought to let the developer iterate faster while developing but is not intended to be used in production environment.

Reviewer: See above comment and adjust accordingly

Author: What is exactly that I have to do?

Reviewer: I think I rewrote the phrase "For this you can use the Flow's `restart` argument to restart the Flow with the updated configuration every time you change the YAML configuration." (or a very similar phrase) in prior commit suggestion.

````

````{admonition} Note
:class: note
This feature requires watchfiles>=0.18 package to be installed.
````

> To see how this would work, let's have a Flow defined in a file `flow.yml` with restart option.

Reviewer: See prior comments on similar text

Reviewer: Apply similar comments above to whole file

Author: and here?
